File: learning-akka/akkademy-server/src/test/java/book/java/akkademy/AkkademyDbTest.java
package book.java.akkademy;

import akka.actor.ActorRef;
import akka.actor.ActorSystem;
import akka.actor.Props;
import akka.testkit.TestActorRef;
import book.kotlin.messages.SetRequest;
import org.junit.AfterClass;
import org.junit.BeforeClass;
import org.junit.Test;

import static org.junit.Assert.assertEquals;

public class AkkademyDbTest {

	private static ActorSystem system;

	@BeforeClass
	public static void setUp() {
		system = ActorSystem.create();
	}

	@AfterClass
	public static void shutDown(){
		system.shutdown();
	}

	@Test
	public void itShouldPlaceKeyValueFromSetMessageIntoMap() {
		TestActorRef<AkkademyDb> actorRef = TestActorRef.create(system, Props.create(AkkademyDb.class));
		actorRef.tell(new SetRequest("key", "value"), ActorRef.noSender());
		AkkademyDb akkademyDb = actorRef.underlyingActor();
		assertEquals(akkademyDb.map.get("key"), "value");
	}

}

File: learning-akka/akkademy-server/src/test/kotlin/book/kotlin/akkademy/AkkademyDbTest.kt
package book.kotlin.akkademy

import akka.actor.ActorSystem
import book.kotlin.messages.SetRequest
import book.kotlin.util.TestActorRef
import book.kotlin.util.tell
import org.jetbrains.spek.api.Spek
import org.jetbrains.spek.api.dsl.describe
import org.jetbrains.spek.api.dsl.it
import org.jetbrains.spek.subject.SubjectSpek
import org.junit.Assert.assertEquals
import org.junit.platform.runner.JUnitPlatform
import org.junit.runner.RunWith

@RunWith(JUnitPlatform::class)
class AkkademyDbTest : Spek({

	var system: ActorSystem? = null

	beforeGroup {
		system = ActorSystem.create()
	}

	afterGroup {
		system!!.shutdown()
	}

	describe("akkademyDb") {
		describe("given SetRequest") {
			it("should place key/value into map") {
				val actorRef = TestActorRef<AkkademyDb>(system!!)
				actorRef.tell(SetRequest("key", "value"))
				val akkademyDb = actorRef.underlyingActor()
				assertEquals(akkademyDb.map["key"], "value")
			}
		}
	}

})

